scm-step/src/main/java/org/jenkinsci/plugins/workflow/steps/scm/GitStep.java

```diff
@@ -78,6 +119,58 @@ public DescriptorImpl() {
             GitSCM.class.hashCode();
         }
 
+        // copy/paste from GitSCM.DescriptorImpl
+
+        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item project,
+                                                     @QueryParameter String url) {
+            if (project == null || !project.hasPermission(Item.CONFIGURE)) {
+                return new StandardListBoxModel();
+            }
+            return new StandardListBoxModel()
+                    .withEmptySelection()
+                    .withMatching(
+                            GitClient.CREDENTIALS_MATCHER,
+                            CredentialsProvider.lookupCredentials(StandardCredentials.class,
+                                    project,
+                                    ACL.SYSTEM,
+                                    GitURIRequirementsBuilder.fromUri(url).build())
+                    );
+        }
+
+        public FormValidation doCheckUrl(@AncestorInPath Item project,
+                                         @QueryParameter String credentialsId,
+                                         @QueryParameter String value) throws IOException, InterruptedException {
+
+            if (project == null || !project.hasPermission(Item.CONFIGURE)) {
+                return FormValidation.ok();
+            }
+
+            String url = Util.fixEmptyAndTrim(value);
+            if (url == null)
+                return FormValidation.error("Please enter Git repository.");
+
+            if (url.indexOf('$') >= 0)
+                // set by variable, can't validate
+                return FormValidation.ok();
```

Comment:
This block can be deleted for workflows, as variable substitution only makes sense for freestyle projects. (Even if the URL is computed in the flow script, the `@DataBoundConstructor` will get the evaluated result, and anyway this form validation is only for the snippet generator, where you should only be entering a _sample_ URL.)